batched/dense/unit_test/Test_Batched_SerialPbtrs.hpp

```diff
@@ -0,0 +1,295 @@
+      EXPECT_NEAR_KK(h_x0(ib, i), h_x_ref(ib, i), eps);
+    }
+  }
+}
+
+template <typename DeviceType, typename ScalarType, typename LayoutType, typename ParamTagType, typename AlgoTagType>
+/// \brief Implementation details of batched pbtrs test
+///        Confirm A * x = b, where
+///
+/// \param N [in] Batch size of RHS (banded matrix can also be batched matrix)
+/// \param k [in] Number of superdiagonals or subdiagonals of matrix A
+/// \param BlkSize [in] Block size of matrix A
+void impl_test_batched_pbtrs(const int N, const int k, const int BlkSize) {
+  using ats        = typename Kokkos::ArithTraits<ScalarType>;
+  using RealType   = typename ats::mag_type;
+  using View2DType = Kokkos::View<ScalarType **, LayoutType, DeviceType>;
+  using View3DType = Kokkos::View<ScalarType ***, LayoutType, DeviceType>;
+
+  View3DType A("A", N, BlkSize, BlkSize), A_reconst("A_reconst", N, BlkSize, BlkSize);
+  View3DType Ab("Ab", N, k + 1, BlkSize);                                             // Banded matrix
+  View2DType x0("x0", N, BlkSize), x_ref("x_ref", N, BlkSize), y0("y0", N, BlkSize);  // Solutions
+
+  using execution_space = typename DeviceType::execution_space;
+  Kokkos::Random_XorShift64_Pool<execution_space> rand_pool(13718);
+  ScalarType randStart, randEnd;
+
+  // Initialize A_reconst with random matrix
+  KokkosKernels::Impl::getRandomBounds(1.0, randStart, randEnd);
+  Kokkos::fill_random(A, rand_pool, randStart, randEnd);
+  Kokkos::fill_random(x0, rand_pool, randStart, randEnd);
+  Kokkos::deep_copy(x_ref, x0);
+
+  // Make the matrix Positive Definite Symmetric and Diagonal dominant
+  random_to_pds(A, A_reconst);
+  Kokkos::deep_copy(A, ScalarType(0.0));
+
+  // Create banded triangluar matrix in normal and banded storage
+  using ArgUplo = typename ParamTagType::uplo;
+  create_banded_pds_matrix<View3DType, View3DType, ArgUplo>(A_reconst, A, k, false);
+
+  create_banded_triangular_matrix<View3DType, View3DType, ArgUplo>(A_reconst, Ab, k, true);
+
+  Kokkos::fence();
+
+  // Factorize with Pbtrf: A = U**H * U or A = L * L**H
+  Functor_BatchedSerialPbtrf<DeviceType, View3DType, ParamTagType, AlgoTagType>(Ab).run();
+
+  // pbtrs (Note, Ab is a factorized matrix of A)
+  auto info = Functor_BatchedSerialPbtrs<DeviceType, View3DType, View2DType, ParamTagType, AlgoTagType>(Ab, x0).run();
+
+  Kokkos::fence();
+  EXPECT_EQ(info, 0);
+
+  // Gemv to compute A*x0, this should be identical to x_ref
+  Functor_BatchedSerialGemv<DeviceType, ScalarType, View3DType, View2DType, View2DType>(1.0, A, x0, 0.0, y0).run();
+
+  Kokkos::fence();
+
+  // this eps is about 10^-14
+  RealType eps = 1.0e3 * ats::epsilon();
+
+  auto h_y0    = Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), y0);
+  auto h_x_ref = Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), x_ref);
+
+  // Check A * x0 = x_ref
+  for (int ib = 0; ib < N; ib++) {
+    for (int i = 0; i < BlkSize; i++) {
+      EXPECT_NEAR_KK(h_y0(ib, i), h_x_ref(ib, i), eps);
```

Comment:
Same as above, these values should be checked with a relative error computation.